src/common.ts

```diff
@@ -243,3 +254,69 @@ function translateData(responseType: string | undefined, data: any) {
       return data;
   }
 }
+
+/**
+ * An experimental error redactor.
+ *
+ * @param config Config to potentially redact properties of
+ * @param response Config to potentially redact properties of
+ *
+ * @experimental
+ */
+export function defaultErrorRedactor(data: {
+  config?: GaxiosOptions;
+  response?: GaxiosResponse;
+}) {
+  const REDACT =
+    '<<REDACTED> - See `errorRedactor` option in `gaxios` for configuration>.';
+
+  function redactHeaders(headers?: Headers) {
+    if (headers && 'Authentication' in headers) {
+      headers['Authentication'] = REDACT;
+    }
+  }
+
+  function redactString(obj: GaxiosOptions, key: keyof GaxiosOptions) {
+    if (
+      typeof obj === 'object' &&
+      obj !== null &&
+      typeof obj[key] === 'string'
+    ) {
+      const text = obj[key];
+
+      if (/grant_type=/.test(text) || /assertion=/.test(text)) {
+        obj[key] = REDACT;
```

Comment:
I don't know this code well enough; I'm a little bit afraid that without making a deep copy, we might be breaking some of the configuration that would possibly affect the subsequent calls. I'm not sure if this is the case though. Do you think having a unit test that would assign some configuration to the global instance, then make a failing request, then make a successful request against `nock` might make sense here?

Reply:
Great point - I shallow-copied the objects (as the redacted props are on the top-level, fortunately) and created specific interfaces for the redact-able properties. Working on tests now

Reply:
Added tests as well